syntax = "proto3";
package plugin_version;




import "model/agent_admin/plugin.proto";
import "model/agent_admin/plugin_version.proto";

/*
ListPluginVersionV1请求
*/
message ListPluginVersionV1Request {
    /*
    组件ID
    */
    string id = 1 ;
    /*
    页码
    */
    int32 page = 2 ;
    /*
    每页数据量
    */
    int32 pageSize = 3 ;
}

/*
ListPluginVersionV1返回
*/
message ListPluginVersionV1Response {
message List {
    /*
    插件包版本名称
    */
    string versionName = 1 ;
    /*
    所属插件包
    */
    agent_admin.Plugin agentPlugin = 2 ;
    /*
    基线版本
    */
    agent_admin.PluginVersion baseVersion = 3 ;
    /*
    ID
    */
    string id = 4 ;
    /*
    仓库版本ID
    */
    string repoVersionId = 5 ;
    /*
    版本名称
    */
    string name = 6 ;
    /*
    备注
    */
    string memo = 7 ;
    /*
    所属插件ID
    */
    string pluginId = 8 ;
    /*
    创建者
    */
    string creator = 9 ;
    /*
    创建时间
    */
    int32 ctime = 10 ;
    /*
    修改时间
    */
    int32 mtime = 11 ;
}
        
    
    /*
    页数
    */
    int32 page = 1 ;
    /*
    页大小
    */
    int32 page_size = 2 ;
    /*
    总数
    */
    int32 total = 3 ;
    /*
    数据列表
    */
    repeated List list = 4 ;
}

/*
ListPluginVersionV1Api返回
*/
message ListPluginVersionV1ResponseWrapper {
    /*
    返回码
    */
    int32 code = 1 ;
    /*
    返回码解释
    */
    string codeExplain = 2 ;
    /*
    错误详情
    */
    string error = 3 ;
    /*
    返回数据
    */
    plugin_version.ListPluginVersionV1Response data = 4 ;
}

